src/Mvc/Mvc.Core/src/Diagnostics/MvcDiagnostics.cs

```diff
@@ -0,0 +1,709 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc.Abstractions;
+using Microsoft.AspNetCore.Mvc.Filters;
+using Microsoft.AspNetCore.Routing;
+
+namespace Microsoft.AspNetCore.Mvc.Diagnostics
+{
+    public sealed class BeforeAction : MvcDiagnostic
```

Comment:
We typically do one class per-file - but I don't feel really strongly about it in this case.